frontend/src/components/MainSearch/MainSearch.js

```diff
@@ -8,14 +9,28 @@ class MainSearch extends Component {
     this.state = {
       searchTerm: '',
       error: '',
-      open: false,
+      open: true,
     };
 
     this.handleChange = this.handleChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
     this.togglePanel = this.togglePanel.bind(this);
   }
 
+  componentWillMount() {
```

Comment:
I believe `componentWillMount` is considered unsafe, and we should just use `componentDidMount`. I might consider setting `open` to false by default, and setting it to `true` if the cookie is missing. I might also set the `visit` cookie in `componentDidMount`, as well, after you check it, and then we don't have to worry about `componentWillUnmount`.